llmfoundry/common/convert_dataset.py

```diff
@@ -300,6 +302,11 @@ def build_hf_dataset(dataset_name: str,
                                   data_subset=data_subset,
                                   split=split)
     else:
+        if not isinstance(tokenizer, PreTrainedTokenizerBase):
+            raise ValueError(
+                f'{tokenizer=} must be of type PreTrainedTokenizerBase')
+        if max_length == 0:
+            raise ValueError(f'max_length must be set.')
```

Comment:
In this case, can we just not have it be a default arg of 0 and make it a required arg

Reply:
if we take the `if mode == ConcatMode.NO_CONCAT:` path, then max_length is not required.

Reply:
small nit: in this case, its common to do `max_length: Optional[int] = None` , and then check for `None`. Avoids using a magic number (0), and makes the usage clear in the type annotation. Also noticed that `max_length` is missing from the docstrings.